packages/styled-components/src/sheet/GroupIDAllocator.js

```diff
@@ -19,7 +19,12 @@ export const getGroupForId = (id: string): number => {
     return (groupIDRegister.get(id): any);
   }
 
-  const group = nextFreeGroup++;
+  while (reverseRegister.has(nextFreeGroup)) {
+    nextFreeGroup++;
+  }
+
+  const group = nextFreeGroup;
```

Comment:
This is technically more correct since we know that this group will be taken, so there's no need to check the reverse register on the next run against it.

```suggestion
  const group = nextFreeGroup++;
```

Reply:
You're right! I fixed it :)